### PR TITLE
Active report should be visible in sidebar if mode is GSD

### LINKED ISSUE #1682: When GSD priority setting is enabled continue to show current chat in LHN while viewing the chat
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___
Upwork listing: https://www.upwork.com/jobs/~0110ec2c11a21481f0

## Expected Result:
When GSD priority setting is enabled, chat should show in Left Hand Navigation pane while you are viewing it 

## Actual Result:
Chat disappears from Left Hand Navigation pane when you view it

## Action Performed:
1. Open e.cash
2. Ensure you have a couple unread chats in Left Hand Navigation (LHN)
3. Click your user icon - click dropdown to enable GSD Priority Mode
4. Observe unread chats in LHN
5. click one, start typing.
6. Observe the chat disappear from LHN while you're composing/viewing it 

## Platform: 

Desktop App 
 
**Version Number:** Version 1.0.2-12 (v1.0.2-12)

**Notes/Photos/Videos:** 

Viewing Mitch's Settings, notice GSD Priority Mode has been selected and Rafe's unread message in Mitch's Left Hand Navigation pane:
Screen Shot 2021-03-09 at 8 47 45 AM

Viewing Mitch's direct message with Rafe, notice its disappearance from Left Hand Navigation:
Screen Shot 2021-03-09 at 8 50 03 AM

### PR DESCRIPTION
**Pull request by parasharrajat**
Please Review.

### Details
We forget to set the Check for Pinned and Active report while adjusting the SiebarOptions for the GSD mode. I have fixed it and Make the if statement more readable.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1682 

### Tests
 1. open any chat.
 2. Now Set the Settings Preioty Mode to _GSD_.
 3. Active report should be visible in the sidebar.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots

https://user-images.githubusercontent.com/24370807/110586774-b2d13800-8198-11eb-9d9e-fbbefe1991e5.mp4

### CODE REVIEW

**Comment by parasharrajat** [comment]
I may have found some strange behavior.  While the NewChat Menu is open, try to press rapidly in the backdrop area. The menu shows open and closed behavior. I think it should just close and the next click should work on the element that is behind the Cursor.

https://user-images.githubusercontent.com/24370807/110588136-9df5a400-819a-11eb-828f-343d138e8335.mp4

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -185,13 +185,10 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
...
-        const shouldFilterReport = !showReportsWithNoComments && hasNoComments
-            && report.reportID !== activeReportID && !report.isPinned;
-        if (shouldFilterReport) {
-            return;
-        }
-        if (hideReadReports && report.unreadActionCount === 0 && !report.isPinned) {
+        const shouldFilterReportIfEmpty = !showReportsWithNoComments && report.lastMessageTimestamp === 0;
+        const shouldFilterReportIfRead = hideReadReports && report.unreadActionCount === 0;
+        const shouldFilterReport = shouldFilterReportIfEmpty || shouldFilterReportIfRead;
+        if (report.reportID !== activeReportID && !report.isPinned && shouldFilterReport) {
```
Comment (Jag96) [inline src/libs/OptionsListUtils.js:191]: The logic here looks good and the tests pass! However, with this new logic it seems the comment above the const declarations is out of date (it doesn't take into account pinned reports or gsd mode). IMO the const names and logic read easily, so you can just remove that comment.

**Review by Jag96 — CHANGES_REQUESTED** [review]
Just one request to clean up the old comment, otherwise looks good.

As for the new issue you've found, you can refer to [this section of the contributing guidelines](https://github.com/Expensify/Expensify.cash/blob/master/CONTRIBUTING.md#filing-issues) to create a new issue. Once you create the issue, feel free to post your proposed solution as a comment and somebody will review the issue.

File: src/libs/OptionsListUtils.js (hunk 1 above)
Comment (parasharrajat) [reply src/libs/OptionsListUtils.js:195]: I have updated the comments. Let me know if they look fine. Let's keep some sorts of comments as in the future we could need more cases.

**Review by Jag96 — CHANGES_REQUESTED** [review]

File: src/libs/OptionsListUtils.js [hunk 2]
```diff
@@ -183,15 +183,16 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
...
-        // Skip this entry if it has no comments and is not the active report. We will only show reports from
-        // people we have sent or received at least one message with.
-        const hasNoComments = report.lastMessageTimestamp === 0;
-        const shouldFilterReport = !showReportsWithNoComments && hasNoComments
-            && report.reportID !== activeReportID && !report.isPinned;
-        if (shouldFilterReport) {
-            return;
-        }
-        if (hideReadReports && report.unreadActionCount === 0 && !report.isPinned) {
+        // Skip this entry only if the report is not an active report & is not Pinned
```
[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/libs/OptionsListUtils.js:186]: Ok. Let's just remove all the comments as they are self-explanatory via variable names. What do you say @Jag96

[the remark this answers is not in the document]
Comment (Jag96) [reply src/libs/OptionsListUtils.js:186]: Sounds good 👍

**Comment by parasharrajat** [comment]
@Jag96  Here we go. Updated. Please review.

**Review by Jag96 — APPROVED** [review]
LGTM

**Pull request merged**